### ISSUE TITLE
Allow .txt file extension for plain text robot files

### ISSUE THREAD

**Issue by conclavia** [tomi/vscode-rf-language-server#10]
Currently the only files that are parsed are `.robot` for plain text and `.py` for python. If any files with other extensions are picked up by the includePaths patterns then it looks like they get filtered out.

```
const allowedFileExts = new Set(
  Config.getHasPythonKeywords() ? [".robot", ".py"] : [".robot"]
);

...

const filePaths = files
  // User can configure patterns that include other files than .robot.
  // Filter those out.
  .filter(fileUri => isAllowedFile(fileUri))
  .map(fileUri => fileUri.fsPath);
```

Some RF projects also use the alternative `.txt` extension for plain text robot files (particularly older projects). It would be great if these could be detected and parsed as well, either through a configuration option for allowed file extensions or by not filtering the includePaths results and just using includePaths/excludePaths to filter if needed.

**Comment by tomi** [comment]
Thank you for creating an issue! Yes I think we could allow also `.txt` files to be parsed as it is an allowed file extension for plain text format robot files. This allows the language server to become aware of the keywords and variables defined in `.txt` files.

There is however the issue that VSCode won't recognize the `.txt` files to be robot files, so syntax highlighting, completion suggestions or other editing features won't be available when editing those files. There are 3 static (i.e. we can't use a user configurable option) ways how to configure for VSCode to recognize a file to belong to certain language (as documented [here](https://code.visualstudio.com/docs/extensionAPI/extension-points#_contributeslanguages)):

1. By file extension. This is what we use now (`.robot`). I'm reluctant to add `.txt` here, as it's usually associated with plain text files.
2. By filename. We could use for example `.robot.txt`, but that's not how the files are named and if you need to change the filename, you might as well rename them to `.robot`
3. First line inside the file. We could try to recognize a table (e.g. `*** Settings ***`). but it's not reliable if you have for example comments at the beginning of the file.

**Comment by conclavia** [comment]
Thanks, @tomi! You raise a very good point about the language recognition. Globally treating `.txt` files as robot framework would not be ideal :-)

At the moment I'm using a workspace setting to force VSCode to recognise the `.txt` extension as robot for this specific project:
```
"files.associations": {
    "*.txt": "robot"   
}
```
That gives me robot syntax highlighting for my `.txt` files. I can see from the logs that it is also triggering a search for completion suggestions, although without any results. I was hoping that if we allowed the language server to parse `.txt` files then it would be enough to get intellisense working as well.

Allowing the user to configure the association with a workspace- or user-level setting rather than having the extension control it seems like a good compromise. That way we can treat the files differently in different projects as needed. In the rare case where there might be a plain `.txt` file in my robot workspace I can always just manually switch the language once the file is open.

What do you think?

**Comment by tomi** [comment]
Oh nice, I wasn't aware that you can associate files like that! Yes I think this is totally doable. I'll add support for `.txt` files as soon as I have time. It's probably better to leave it so that by default only `.robot` files are parsed, but you can add a pattern to include `.txt` as well:

```json
"rfLanguageServer.includePaths": [
    "**/*.robot",
    "**/*.txt"
]
```

**Comment by tomi** [comment]
@conclavia Please check out the newly released version 2.0.0. It includes a support for `.txt` files.
